#! /bin/sh -e
## 18_fix_makeqpf_qembed.dpatch by Martin Loschwitz <madkiss@madkiss.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix makeqpf by make it not use stuff defined in Qt embedded

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)  
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -ruN qt-x11-free-3.3.3.old/tools/makeqpf/main.cpp qt-x11-free-3.3.3/tools/makeqpf/main.cpp
--- qt-x11-free-3.3.3.old/tools/makeqpf/main.cpp	2004-06-23 14:13:52.000000000 +0000
+++ qt-x11-free-3.3.3/tools/makeqpf/main.cpp	2004-08-14 08:53:01.000000000 +0000
@@ -124,8 +124,6 @@
     }
 };
 
-extern QString qws_topdir();
-
 class MakeQPF : public QMainWindow
 {
     Q_OBJECT
@@ -139,7 +137,7 @@
 	view->addColumn("Weight");
 	view->addColumn("Style");
 	setCentralWidget(view);
-	QString fontdir = qws_topdir() + "/lib/fonts";
+	QString fontdir = QString(getenv("QTDIR")) + "/lib/fonts";
 	readFontDir(fontdir);
 
 	connect(view,SIGNAL(selectionChanged(QListViewItem*)),
